sn_networking/src/record_store.rs

```diff
@@ -657,14 +661,33 @@ impl RecordStore for ClientRecordStore {
     fn remove_provider(&mut self, _key: &Key, _provider: &PeerId) {}
 }
 
-// Using a linear growth function, and be tweaked by `received_payment_count`
-// to allow nodes receiving too many replication copies can still got paid.
-fn calculate_cost_for_records(step: usize, received_payment_count: usize) -> u64 {
-    use std::cmp::max;
-
-    let ori_cost = (10 * step) as u64;
-    let divider = max(1, step / max(1, received_payment_count)) as u64;
-    max(10, ori_cost / divider)
+// Using a linear growth function, and be tweaked by `received_payment_count` and `max_records`,
+// to allow nodes receiving too many replication copies can still got paid,
+// and gives an exponential pricing curve when storage reaches high.
+fn calculate_cost_for_records(
```

Comment:
For calculations that are going to be important, we should enforce results with a simple unit test.